pineappl/src/boc.rs

```diff
@@ -3,13 +3,63 @@
 //!
 //! [`Grid`]: super::grid::Grid
 
+use bitflags::bitflags;
 use float_cmp::approx_eq;
 use itertools::Itertools;
 use serde::{Deserialize, Serialize};
 use std::cmp::Ordering;
 use std::str::FromStr;
 use thiserror::Error;
 
+bitflags! {
+    /// TODO
+    #[derive(Clone, Copy, Deserialize, PartialEq, Serialize)]
+    #[repr(transparent)]
+    pub struct Scale: u32 {
+        /// TODO
+        const REN = 0b001;
+        /// TODO
+        const FAC = 0b010;
+        /// TODO
+        const FRG = 0b100;
+    }
+}
+
+/// TODO
+#[derive(Clone, Copy, Deserialize, PartialEq, Serialize)]
+pub enum Kinematics {
+    /// TODO
+    Mu2(Scale),
+    /// TODO
+    X(usize),
+}
+
+impl Kinematics {
+    /// TODO
+    pub const MU2_R: Self = Self::Mu2(Scale::REN);
+
+    /// TODO
+    pub const MU2_F: Self = Self::Mu2(Scale::FAC);
```

Comment:
since you are already on it: you may want to consider an even more general case: there is µ for every convolution and not per convolution type. E.g. consider Unpol x Pol x Timelike - there are in principle three different mus, two of which one could call a factorization scale. However, there is no pQCD argument which forces them to be the same, although they are often _chosen_ to be the same. The same holds for unequal hadron collision, e.g. p-A at LHC: in principle I can choose an individual scale for each beam. For equal beams, e.g. p-p, this does not hold, as here things are symmetric by definition. (Of course the ren. scale is always there and separate)

Reply:
That's a good suggestion, and I had something even more general in mind which is that you separately set not only the scale of each convolution function (PDF, FF, ..), but also the scale of different couplings (alphas, scale-dependent alpha, [other couplings in MS-bar renormalized EFTs](https://github.com/NNPDF/pineappl/issues/98), ...). But all of this requires a bit more thinking which will have to be done afterwards, with lower priority. The highest priority for the time being is to get this branch merged and to make 3 convolutions working.